File: frontend/src/services/ReviewService.js
import api from "./api";

export const createReview = async (reviewData) => {
    const response = await api.post("/reviews", reviewData);
    return response.data;
};

export const getReviewsByProduct = async (productId) => {
    const response = await api.get(`/reviews/product/${productId}`);
    return response.data;
};

export const getReviewsByBuyer = async (buyerId) => {
    const response = await api.get(`/reviews/buyer/${buyerId}`);
    return response.data;
};

export const deleteReview = async (reviewId) => {
    await api.delete(`/reviews/${reviewId}`);
};

File: src/main/java/com/ebizzness/ecommerce/service/ReviewService.java
package com.ebizzness.ecommerce.service;

import com.ebizzness.ecommerce.dto.request.ReviewRequest;
import com.ebizzness.ecommerce.dto.response.ReviewResponse;
import java.util.List;

public interface ReviewService {

    ReviewResponse createReview(ReviewRequest request);

    List<ReviewResponse> getReviewsByProduct(Long productId);

    List<ReviewResponse> getReviewsByBuyer(Long buyerId);

    void deleteReview(Long reviewId);
}
